kiota_abstractions/request_information.py

```diff
@@ -291,17 +291,26 @@ def _set_content_and_content_type_header(
             self.headers.try_add(self.CONTENT_TYPE_HEADER, content_type)
         self.content = writer.get_serialized_content()
 
-    def _replace_enum_values_with_string_representation(self, value: Any) -> Any:
+    def _get_sanitized_value(self, value: Any) -> Any:
         """Replaces enum values with their string representation.
 
         Args:
             value (Any): The value to replace.
         """
+        sanitized_value = value
         if isinstance(value, Enum):
-            return value.value
-        if isinstance(value, list) and all(isinstance(x, Enum) for x in value):
-            return ','.join([x.value for x in value])
-        return value
+            sanitized_value = value.value
+        elif isinstance(value, list) and all(isinstance(x, Enum) for x in value):
+            sanitized_value = ','.join([x.value for x in value])
+        elif isinstance(value, UUID):
+            sanitized_value = str(value)
+        elif isinstance(value, datetime):
+            sanitized_value = value
+        elif isinstance(value, date):
+            sanitized_value = str(value)
+        elif isinstance(value, time):
+            sanitized_value = str(value)
```

Comment:
```suggestion
        elif any([isinstance(value, UUID), isinstance(value, date), isinstance(value, time)]:)
            sanitized_value = str(value)
        elif isinstance(value, datetime):
            sanitized_value = value
```
Collapse the `str`ingers.